### PR TITLE
Implement NaT properties/methods directly

### PR DESCRIPTION
**Pull request by jbrockmendel**
Causes build errors, referenced in #17756 

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/scalar/test_nat.py [hunk 1]
```diff
@@ -156,6 +158,44 @@ def test_NaT_methods():
     assert NaT.isoformat() == 'NaT'
 
 
+def test_NaT_docstrings():
```
Comment (jreback) [inline pandas/tests/scalar/test_nat.py]: add the issue number as a comment

File: pandas/tests/scalar/test_nat.py [hunk 2]
```diff
@@ -156,6 +158,44 @@ def test_NaT_methods():
...
+    nat_names = dir(NaT)
+
+    ts_names = dir(Timestamp)
+    ts_missing = [x for x in ts_names if x not in nat_names and
+                  not x.startswith('_')]
+    ts_missing.sort()
+    ts_expected = ['freqstr', 'normalize', 'offset',
+                   'to_julian_date', 'to_period', 'tz']
+    assert ts_missing == ts_expected
+
```
Comment (jreback) [inline pandas/tests/scalar/test_nat.py]: give a comment before each 'section' of the test so a future reader gets what you are testing

**Comment by jbrockmendel** [comment]
Just added comments.  Note that the test implemented here is identical (except for the new comments) to the one implemented in #17756

File: pandas/_libs/tslib.pyx [hunk 3]
```diff
@@ -862,6 +890,90 @@ class NaTType(_NaT):
...
+    strftime = _make_error_func('strftime', datetime)
+    isocalendar = _make_error_func('isocalendar', datetime)
+    dst = _make_error_func('dst', datetime)
+    ctime = _make_error_func('ctime', datetime)
+    time = _make_error_func('time', datetime)
+    toordinal = _make_error_func('toordinal', datetime)
+    tzname = _make_error_func('tzname', datetime)
+    utcoffset = _make_error_func('utcoffset', datetime)
+
+    # Timestamp has empty docstring for some methods.
```
Comment (jreback) [inline pandas/_libs/tslib.pyx]: these are not defined directly in ``Timestamp/_Timestamp`` rather directly in ``datetime.datetime``, maybe direct to there? (I don't know where its actually defined, nor why the inheritence messes this up)

**Review by jreback — APPROVED** [review]
lgtm. small comment (can defer this unless its an easy fix; if defering can you make an issue / put on todo list)

File: pandas/_libs/tslib.pyx (hunk 3 above)
Comment (jbrockmendel) [reply pandas/_libs/tslib.pyx]: They are defined in Timestamp and have empty docstrings there.  No idea why.  Will add to todo list.

Comment (jreback) [reply pandas/_libs/tslib.pyx]: ok great.

**Comment by jreback** [comment]
thanks!

**Pull request merged**